flow360/component/simulation/outputs/output_entities.py

```diff
@@ -141,3 +139,42 @@ class PointArray(_PointEntityBase):
     start: LengthType.Point = pd.Field(description="The starting point of the line.")
     end: LengthType.Point = pd.Field(description="The end point of the line.")
     number_of_points: int = pd.Field(gt=2, description="Number of points along the line.")
+
+
+class PointArray2D(_PointEntityBase):
+    """
+    :class:`PointArray2D` class for defining multiple equally spaced points along the u and
+    v axes of a trapezoid.
+
+
+    Example
+    -------
+    Define :class:`PointArray2D` with points equally distributed on a trapezoid with
+    origin (1.0, 0.0, 0.0) * fl.u.m. There are 7 equally spaced points along the trapezoid's u-axis
+    of (0.5, 1.0, 0.2) * fl.u.m and 10 equally spaced points along the its v-axis of
+    (0.1, 0, 1) * fl.u.m.
+
+    Both the starting and end points are included in the :class:`PointArray`.
+
+    >>> fl.PointArray(
+    ...     name="Trapezoid_1",
+    ...     origin=(1.0, 0.0, 0.0) * fl.u.m,
+    ...     u_axis_vector=(0.5, 1.0, 0.2) * fl.u.m,
+    ...     v_axis_vector=(0.1, 0, 1) * fl.u.m,
+    ...     u_number_of_points=7,
+    ...     v_number_of_points=10
+    ... )
+
+    ====
+    """
+
+    private_attribute_entity_type_name: Literal["PointArray2D"] = pd.Field(
+        "PointArray2D", frozen=True
+    )
+    private_attribute_id: str = pd.Field(default_factory=generate_uuid, frozen=True)
+    # pylint: disable=no-member
+    origin: LengthType.Point = pd.Field(description="The origin of the trapezoid.")
+    u_axis_vector: LengthType.Point = pd.Field(description="The scaled u-axis of the trapezoid.")
+    v_axis_vector: LengthType.Point = pd.Field(description="The scaled v-axis of the trapezoid.")
```

Comment:
LengthType.Point will accept (0, 0, 0) which might be technically correct but I believe it doesn't make sense.

```suggestion
    u_axis_vector: LengthType.Axis = pd.Field(description="The scaled u-axis of the trapezoid.")
    v_axis_vector: LengthType.Axis = pd.Field(description="The scaled v-axis of the trapezoid.")
```